### ISSUE TITLE
can't create nippy with datasets as column

### ISSUE THREAD

**Issue by genmeblog** [techascent/tech.ml.dataset#122]
```clojure
(def d (ds/->dataset {:a [1 2]
                      :datasets [(ds/->dataset [{:a 1}])
                                 (ds/->dataset [{:b 2}])]}))

;; => _unnamed [2 2]:
;;    | :a |       :datasets |
;;    |----|-----------------|
;;    |  1 | _unnamed [1 1]: |
;;    |    |                 |
;;    |    | \| :a \|        |
;;    |    | \|----\|        |
;;    |    | \|  1 \|        |
;;    |  2 | _unnamed [1 1]: |
;;    |    |                 |
;;    |    | \| :b \|        |
;;    |    | \|----\|        |
;;    |    | \|  2 \|        |

(tech.io/put-nippy! "file.nippy" d)


;; exception

1. Unhandled clojure.lang.ExceptionInfo
Unfreezable type: class [Ljava.lang.Object;
                         {:type [Ljava.lang.Object;,
                                 :as-str
                                 "#object[\"[Ljava.lang.Object;\" 0x3730bfa1 \"[Ljava.lang.Object;@3730bfa1\"]"}
                          nippy.clj:  720  taoensso.nippy/throw-unfreezable
                          nippy.clj:  718  taoensso.nippy/throw-unfreezable
```

**Comment by cnuernber** [comment]
Great example.  I was really hoping that would 'just work'...I wonder if object columns can be serialized as persistent vectors instead and if that would work.  Potentially the fix for this is adding a 'dataset' datatype and to have special processing for 'dataset' columns.

**Comment by cnuernber** [comment]
Potentially fixed in 3.07
